packages/domain-events/src/Domain/Event/SubscriptionPurchasedEventPayload.ts

```diff
@@ -2,7 +2,8 @@ import { SubscriptionName } from '@standardnotes/auth'
 
 export interface SubscriptionPurchasedEventPayload {
   userEmail: string
-  subscriptionName: SubscriptionName
+  fromSubscriptionName: SubscriptionName
```

Comment:
from can be undefined right ? - purchasing a subscription from not having one
```suggestion
  fromSubscriptionName?: SubscriptionName
```

Reply:
The fromSubscriptionName in that case should be CorePlan, correct? That way we would know we need to remove the CoreUser role from the user

Reply:
hmm not really ... a basic user: https://github.com/standardnotes/snjs/blob/master/packages/auth/src/Domain/Role/RoleName.ts#L2

can buy a core plan: https://github.com/standardnotes/snjs/blob/master/packages/auth/src/Domain/Subscription/SubscriptionName.ts#L2

so it means that his subscription was null until now.

Reply:
Ah thanks, that clears everything out! For some reason I was confusing Core with Basic thinking they were the same thing. Will update this